### PR TITLE
Add some guidelines for writing CLI endpoints

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
Sorry added another quick bullet point

**Review by nilsdeppe — DISMISSED** [review]

File: docs/DevGuide/PythonBindings.md [hunk 1]
```diff
@@ -150,3 +150,50 @@ See \ref spectre_using_python "Using SpECTRE's Python"
...
+  taking many H5 volume data files as arguments instead of the directory that
+  contains them. This helps with operating on H5 files in segments or other
+  subdirectory structures. Passing many files to a script is easy for the user
+  by using a glob (note: don't take the glob as a string argument, take the
+  expanded list of files directly using `click.argument(..., nargs=-1,
+  type=click.Path(...))`).
+- Never overwrite or delete files without prompting the user or asking them to
+  run with `--force`.
+- When the input to a script is empty, [gracefully decrage to a
+  noop](https://click.palletsprojects.com/en/8.1.x/arguments/#variadic-arguments).
```
Comment (knelli2) [inline docs/DevGuide/PythonBindings.md]: What does `decrage` mean? I've never seen it used. Also, is it noop or no-op?

Comment (nilsvu) [reply docs/DevGuide/PythonBindings.md]: Sorry it's not a fancy word but just a typo. See the link to click.

Comment (knelli2) [reply docs/DevGuide/PythonBindings.md]: Ahh ok makes sense now. Could you fix the typo though?

Comment (nilsvu) [reply docs/DevGuide/PythonBindings.md]: Oh forgot to push

**Review by knelli2 — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**